cmd/puppeth/wizard_genesis.go

```diff
@@ -58,335 +57,175 @@ func (w *wizard) makeGenesis() {
-		genesis.Alloc[common.HexToAddress(common.BlockSigners)] = core.GenesisAccount{
-			Balance: big.NewInt(0),
-			Code:    code,
-			Storage: storage,
-		}
+	genesis.Validators = candidates
+	genesis.NextValidators = candidates
 
-		// Randomize Smart Contract Code
-		randomizeAddress, _, err := randomizeContract.DeployRandomize(transactOpts, contractBackend)
-		if err != nil {
-			fmt.Println("Can't deploy root registry")
-		}
-		contractBackend.Commit()
-
-		code, _ = contractBackend.CodeAt(ctx, randomizeAddress, nil)
-		storage = make(map[common.Hash]common.Hash)
-		contractBackend.ForEachStorageAt(ctx, randomizeAddress, nil, f)
-		genesis.Alloc[common.HexToAddress(common.RandomizeSMC)] = core.GenesisAccount{
-			Balance: big.NewInt(0),
-			Code:    code,
-			Storage: storage,
-		}
+	validatorCap := new(big.Int)
+	validatorCap.SetString(validatorContract.MinCandidateCap, 10)
+	var validatorCaps []*big.Int
+	genesis.ExtraData = make([]byte, 32+len(candidates)*common.AddressLength+65)
+	for i, signer := range candidates {
+		validatorCaps = append(validatorCaps, validatorCap)
+		copy(genesis.ExtraData[32+i*common.AddressLength:], signer[:])
+	}
 
-		fmt.Println()
-		fmt.Println("Which accounts are allowed to confirm in Team MultiSignWallet?")
-		var teams []common.Address
-		for {
-			if address := w.readAddress(); address != nil {
-				teams = append(teams, *address)
-				continue
-			}
-			if len(teams) > 0 {
-				break
-			}
-		}
-		fmt.Println()
-		fmt.Println("How many require for confirm tx in Team MultiSignWallet? (default = 2)")
-		required = int64(w.readDefaultInt(2))
+	fmt.Println()
+	fmt.Println("How many blocks per epoch? (default = 900)")
+	epochNumber := uint64(w.readDefaultInt(900))
+	genesis.Config.XDPoS.Epoch = epochNumber
+	genesis.Config.XDPoS.RewardCheckpoint = epochNumber
+
+	fmt.Println()
+	fmt.Println("How many blocks before checkpoint need to prepare new set of masternodes? (default = 450)")
+	genesis.Config.XDPoS.Gap = uint64(w.readDefaultInt(450))
 
-		// MultiSigWallet.
-		multiSignWalletTeamAddr, _, err := multiSignWalletContract.DeployMultiSigWallet(transactOpts, contractBackend, teams, big.NewInt(required))
+	fmt.Println()
+	fmt.Printf("What is foundation wallet address? (default = %s)\n", grandMasters[0].Hex())
+	genesis.Config.XDPoS.FoudationWalletAddr = w.readDefaultAddress(grandMasters[0])
+
+	// Validator Smart Contract Code
+	pKey, _ := crypto.HexToECDSA("b71c71a67e1177ad4e901695e1b4b9ee17ae16c6668d313eac2f96dbcda3f291")
+	addr := crypto.PubkeyToAddress(pKey.PublicKey)
+	contractBackend := backends.NewXDCSimulatedBackend(core.GenesisAlloc{addr: {Balance: big.NewInt(1000000000)}}, 10000000, params.TestXDPoSMockChainConfig, nil)
+	transactOpts := bind.NewKeyedTransactor(pKey)
+
+	validatorAddress, _, err := validatorContract.DeployValidator(transactOpts, contractBackend, candidates, validatorCaps, owner, grandMasters, int64(minCandidateThreshold))
```

Comment:
suggest int64(minCandidateThreshold) to uint(minCandidateThreshold)

i dont know golang , but minCandidates is not negative number

Reply:
int64 is abigen generated code, no need to change.